### PR TITLE
Adding @Alias decorator and allow multiple @Expose decorators

### LINKED ISSUE #121: Alias names / Different serialized keynames
**The use case:**
I use this library for a REST API wrapper where I map the returned JSON responses to my classes. I am struggling with just one specific case.

The REST Api returns a `cardName`, but unfortunately no `cardId` which could be used to retrieve the `cardName`. However I can resolve the `cardId` by a given cardName and I highly prefer to store cardIds instead of the cardNames in my database.

**The code:**
```ts
function cardResolver(cardNameOrId: string | number): number {
  // If we serialize a previously deserialized object we will already get the cardId instead of a name
  if (typeof cardNameOrId === 'number') {
    return <number>cardNameOrId;
  }

  return CardHelper.getCardByName(cardNameOrId).id;
}

@Exclude()
export class BaseCard {
  @Transform(cardResolver)
  @Expose({ name: 'name' })
  public id: number;
}
```

**The problem:**
This requirement is a challenge though, because now the library is supposed to handle two different schemas. One is the REST Api which returns the cardName and the other is my database which returns the cardId.

While my `@Transform` function is not an issue (it can handle both json inputs and recognizes the difference using the type), I need some sort of `alias`, since the input json can be either:

```
{ cardName: 'Goblin Barrel' }
```
or

```
{ cardId: 260000013 }
```

**My question:**

Can I achieve this with the given library features or does it make sense to add an alias feature?

### LINKED ISSUE #366: question: `@Expose` decorator with `toClassOnly` in not working as expected
I have that class.
```
Article {
    @Expose({ name: "_id", toClassOnly: true })
    private id: string;
}
```

In comming Json:
```
{
  "_id":"asdasdas"
}
```
I have ```_id```, but I have to return in response Json ```id```, like
```
{
  "id":"asdsdads"
}
```

Function: ```classToPlain(article)``` returns that json:
```
{
  "_id":"asdasdas"
}
```
but when I view variable article, structure are correct.

Why function classToPlain ignoring ```@Expose({ name: "_id", toClassOnly: true })```

### LINKED ISSUE #677: feature: Provide "alias" decorator
### Description

I'd like to propose adding a new field decorator `@Alias` 
In nutshell it's similar to [Mongoose alias](https://mongoosejs.com/docs/guide.html#aliases) : while running `instanceToPlain` properties which have `@Alias({name: "abc"})` decorator are exposed with an alternative property name.
Reverse logic should be applied when running plainToInstance 

example:
```
        class User {
            @Alias({ name: 'fn'})
            firstName: string;
        }

        const instance = new User();
        instance.firstName = 'Alex';

        const plain = instanceToPlain(instance, {
            useAliases: true,
        });

        expect(plain.firstName).toEqual(undefined);
        expect(plain.fn).toEqual('Alex');
```

A use case I have in my company: we store JSON objects in DynamoDB, these JSON objects are usually 3-5 KB and have pretty long property names. Amazon charges for each 1 KB written so would be amazing if we could shorten property names in a centralized way and save some cost.

### PR DESCRIPTION
**Pull request by rgoupil**
## Description
Thank you for this amazing library 🚀

It has been [discussed](https://github.com/typestack/class-transformer/issues/677) several [times](https://github.com/typestack/class-transformer/issues/121) (sometimes with some [confusion](https://github.com/typestack/class-transformer/issues/366)): in some scenarios, being able to rename a property only when importing or exporting it can be useful.

For instance, I am reading rows from a file in a format A sent by a third party, then convert it to an object in my TS code. It might be logged or stored in a file in another format B later on.
One of the field is named `part_or_item_number` in the format A, which is then converted to `partRef` in my JS object.

### TLDR
`plainToInstance` should accept this object:
```json
{
  "part_or_item_number": "part1234"
}
```

The property should be referenced as follow in the code:
```typescript
findPart(myObj.partRef)
```

And `instanceToPlain` should then produce the following:
```json
{
  "partRef": "part1234"
}
```
___

One would quickly suggest to use `@Expose({ name: 'part_or_item_number' })`, however it would also expose the property under the name `part_or_item_number` when using `instanceToPlain`.

Adding the option `toClassOnly: true` is not cutting it either as it would then hide the property when using `instanceToPlain`.

Another suggestion seen in [one of the thread linked earlier](https://github.com/typestack/class-transformer/issues/366) is to use two `@Expose`:
```typescript
...
  @Expose({ name: 'part_or_item_number', toClassOnly: true })
  @Expose({ toPlainOnly: true })
  partRef: string;
...
```
While it's a pretty neat idea, it simply doesn't work in the current state, as the last `@Expose` will always replace the previous ones.
___

This PR introduces two changes:
* Allow the previous sample of code to work as described
* Introduce an `Alias` decorator to reduce boilerplate in that scenario, using `@Expose` behind the scene to reduce duplication

The code above could be rewritten as the following with the same expected behaviour:
```typescript
...
  @Alias({ from: 'part_or_item_number' })
  partRef: string;
...
```

There is also a slight rewriting of the map manipulation code in `MetadataStorage.ts`. I'm fine reverting it but it greatly improved the readability for me.

## Checklist
- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  - `npm run prettier:check` passes
  - `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

### Fixes
fixes #677, fixes #121, fixes #366

### CODE REVIEW

**Comment by ggozlan** [comment]
What's the status? Is there an alternative?
